tensorflow/python/ops/distributions/multinomial.py

```diff
@@ -260,7 +260,8 @@ def _log_prob(self, counts):
 
   def _log_unnormalized_prob(self, counts):
     counts = self._maybe_assert_valid_sample(counts)
-    return math_ops.reduce_sum(counts * math_ops.log(self.probs), -1)
+    logsumexp = math_ops.reduce_logsumexp(self.logits, -1, keep_dims=True)
```

Comment:
this doesn't _seem_ to be the same calculation.  sorry if i'm being thick, can you show how this calculation is equivalent mathematically to the old one?

Reply:
![default](https://user-images.githubusercontent.com/24216379/31052265-5a7fb326-a6b2-11e7-9761-8cdb7e02fd2c.png)
```
from tensorflow.python.ops import math_ops
import tensorflow as tf

with tf.Session() as sess:
    counts = [1., 2., 3.]
    logits = [-1., 2., 3.]
    probs = tf.exp(logits) / tf.reduce_sum(tf.exp(logits))
    
    # The previous one
    print('The previous one', sess.run(math_ops.reduce_sum(counts * math_ops.log(probs), -1)))
    
    # Minus cross entropy
    print('Minus Cross entropy', sess.run(-tf.nn.softmax_cross_entropy_with_logits(labels=counts, logits=logits)))
    
    # The current one
    logsumexp = math_ops.reduce_logsumexp(logits, -1, keep_dims=True)
    print('The current one',sess.run(math_ops.reduce_sum(counts * (logits - logsumexp), -1)))

    
    # Let some of probs be close to zero
    counts = [1., 2., 3.]
    logits = [-1000., 2000., 2000.]
    probs = tf.exp(logits) / tf.reduce_sum(tf.exp(logits))
    
    # The previous one
    print('The previous one',sess.run(math_ops.reduce_sum(counts * math_ops.log(probs), -1)))
    
    # Minus cross entropy
    print('Minus Cross entropy', sess.run(-tf.nn.softmax_cross_entropy_with_logits(labels=counts, logits=logits)))
    
    # The current one
    logsumexp = math_ops.reduce_logsumexp(logits, -1, keep_dims=True)
    print('The current one',sess.run(math_ops.reduce_sum(counts * (logits - logsumexp), -1)))
```
**The above returns:**

('The previous one', -7.9593754)
('Minus Cross entropy', -7.9593763)
('The current one', -7.9593763)
('The previous one', nan)
('Minus Cross entropy', -3004.1587)
('The current one', -3004.1587)